discussion_grading/discussion_grading.py

```diff
@@ -44,35 +153,223 @@ def resource_string(self, path: str) -> str:
+        return score.get("points_earned") if score else 0
+
+    def get_student_item_dict(self) -> dict:
+        """
+        Return dict required by the submissions app.
+
+        This allows creating and retrieving submissions for a particular student.
+
+        Returns:
+            dict: The student item dict.
+        """
+        return {
+            "student_id": get_anonymous_user_id(self.current_user),
+            "course_id": self.block_course_id,
+            "item_id": self.block_id,
+            "item_type": ITEM_TYPE,
+        }
+
+    def set_score(self) -> None:
+        """
+        Set the score for the current user.
+        """
+        set_score(self.submission_uuid, round(self.raw_score * self.weight), self.weight)
+
+    def create_submission(self, user_stats: dict) -> None:
+        """
+        Get the submission for the current user.
+        """
+        submission_data = create_submission(self.get_student_item_dict(), user_stats)
+        self.submission_uuid = submission_data.get("uuid")
+
+    def get_score(self, user_stats: dict) -> int:
+        """
+        Get the grade for the current user based on the grading method and number of participations.
+
+        Args:
+            user_stats (dict): The number of participations for the current user.
+
+        Returns:
+            int: The grade for the current user.
+        """
+        number_of_participations = sum(user_stats.values())
+
+        if number_of_participations >= self.number_of_participations:
+            return 1
+
+        if self.grading_method == DiscussionGradingMethod.MINIMUM_PARTICIPATIONS.name:
+            return int(number_of_participations >= self.number_of_participations)
+        elif self.grading_method == DiscussionGradingMethod.AVERAGE_PARTICIPATIONS.name:
+            return number_of_participations / self.number_of_participations
+        return 0
+
+    def get_user_stats(self) -> dict:
+        """
+        Get the user stats for the current user.
+
+        These stats include the number of:
+            * threads: learner create a post.
+            * responses: learner respond to a post.
+            * replies: learner comment on response.
+
+        If the discussion forum is not enabled or the user
+        stats are not found, a message is returned.
+
+        Example:
+        >>> self.get_user_stats()
+            {
+                "sucess": True,
+                "user_stats": {
+                    "threads": 1,
+                    "responses": 2,
+                    "replies": 3,
+                },
+            }
+
+        Returns:
+            dict: The user stats for the current user or a message.
+        """
+        try:
+            user_stats = get_course_user_stats(self.block_course_id).get("user_stats")
+        except Exception:  # pylint: disable=broad-except
+            return {
+                "sucess": False,
+                "message": _("Discussion forum is not enabled."),
+            }
```

Comment:
If we know that this exception means that the discussion forum is not enabled, why don't we catch that exception instead of Exception?

Reply:
The exception that returns is specific to the `requests` library. For that we should install that library, or I'm not sure how else to do it.

Reply:
What if the call raises other exceptions, and we're catching it here with `Discussion forum is not enabled.` but that's not the error? In the long term, catching a specific exception type is better, or we could also use a exception-specific message like `exc.message` instead.

Reply:
Then, do you see fit to install the library?